src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -1110,6 +1122,229 @@ CHIP_ERROR ConnectivityManagerImpl::GetWiFiVersion(uint8_t & wiFiVersion)
+        const gchar ** keyMgmtsForFree = keyMgmts;
+        uint8_t res                    = 0;
+        for (const gchar * keyMgmtVal = (keyMgmts != nullptr ? *keyMgmts : nullptr); keyMgmtVal != nullptr;
+             keyMgmtVal               = *(++keyMgmts))
+        {
+            if (g_strcasecmp(keyMgmtVal, "wpa-psk") == 0 || g_strcasecmp(keyMgmtVal, "wpa-none") == 0)
+            {
+                res |= (1 << 2); // SecurityType::WPA_PERSONAL
+            }
+            else if (g_strcasecmp(keyMgmtVal, "wpa-eap"))
+            {
+                res |= (kEAP);
+            }
+        }
+        g_variant_unref(keyMgmt);
+        g_free(keyMgmtsForFree);
+        return res;
+    };
+    auto IsNetworkWPA2PSK = [](GVariant * rsn) -> uint8_t {
+        if (rsn == nullptr)
+        {
+            return 0;
+        }
+        GVariant * keyMgmt = g_variant_lookup_value(rsn, "KeyMgmt", nullptr);
+        if (keyMgmt == nullptr)
+        {
+            return 0;
+        }
+        const gchar ** keyMgmts        = g_variant_get_strv(keyMgmt, nullptr);
+        const gchar ** keyMgmtsForFree = keyMgmts;
+        uint8_t res                    = 0;
+        for (const gchar * keyMgmtVal = (keyMgmts != nullptr ? *keyMgmts : nullptr); keyMgmtVal != nullptr;
+             keyMgmtVal               = *(++keyMgmts))
+        {
+            if (g_strcasecmp(keyMgmtVal, "wpa-psk") == 0 || g_strcasecmp(keyMgmtVal, "wpa-psk-sha256") == 0 ||
+                g_strcasecmp(keyMgmtVal, "wpa-ft-psk") == 0)
+            {
+                res |= (1 << 3); // SecurityType::WPA2_PERSONAL
+            }
+            else if (g_strcasecmp(keyMgmtVal, "wpa-eap") == 0 || g_strcasecmp(keyMgmtVal, "wpa-eap-sha256") == 0 ||
+                     g_strcasecmp(keyMgmtVal, "wpa-ft-eap") == 0)
+            {
+                res |= kEAP;
+            }
+            else if (g_strcasecmp(keyMgmtVal, "sae") == 0)
+            {
+                // wpa_supplicant will include "sae" in KeyMgmt field for WPA3 WiFi, this is not included in the wpa_supplicant
+                // document.
+                res |= (1 << 4); // SecurityType::WPA3_PERSONAL
+            }
+        }
+        g_variant_unref(keyMgmt);
+        g_free(keyMgmtsForFree);
+        return res;
+    };
+    auto GetNetworkSecurityType = [IsNetworkWPAPSK, IsNetworkWPA2PSK](WpaFiW1Wpa_supplicant1BSSProxy * proxy) -> uint8_t {
+        std::unique_ptr<GVariant, GVariantDeleter> wpa(g_dbus_proxy_get_cached_property(G_DBUS_PROXY(proxy), "WPA"));
+        std::unique_ptr<GVariant, GVariantDeleter> rsn(g_dbus_proxy_get_cached_property(G_DBUS_PROXY(proxy), "RSN"));
+
+        uint8_t res = IsNetworkWPAPSK(wpa.get()) | IsNetworkWPA2PSK(rsn.get());
+        if (res == 0)
+        {
+            res = 1; // Open
+        }
+        return res & (0x7F);
+    };
+
+    memcpy(result.ssid, ssidStr, ssidLen);
+    memcpy(result.bssid, bssidBuf, bssidLen);
```

Comment:
Bounds should be checked to make sure ssidLen/bssidLen are withing expected bounds.